openapi.yaml

```diff
@@ -3105,17 +2917,62 @@ components:
         submitter:
           $ref: '#/components/schemas/generic-identifier'
 
-    forecast-rating-submission:
+    mutable-rating-proposal:
+      type: object
+      description: |
+        General type for ratings proposal that is mutable over time, such as forecasts and seasonal ratings.  
+        Like the immutable proposal, it includes general fields created by the system.  
+        Filling in these fields before posting may be ignored, or result in a 400 error from TROLIE.  
+        Properties include:
+        * The time the proposal was last updated.  
+        * The ratings provider that owns the proposal.  For mutable proposals such as those for forecasts or seasonal 
+          ratings, the ID of the owning ratings provider is the ID of the proposal itself.  
+      properties:
+        _links: 
+          allOf:
+          - $ref: '#/components/schemas/hal-links-block'
+          - type: object
+            properties:   
+              superseded-by:  
+                $ref: '#/components/schemas/hal-link'
+        last-updated:
+          $ref: '#/components/schemas/timestamp' 
+        ratings-provider:
+          $ref: '#/components/schemas/entity-id'
+
+    
+
+    proposal-status:
+      type: string
+      description: |
+        Enumeration of the possible statuses of a proposal, or parts of a proposal.  Note that submitals of proposals do not
+        set this value.  It is system-generated after submission.  The following values are possible:
+        * *pending* indicates that the system has not yet processed this proposal.  
+        * *accepted* indicates that the proposal has been accepted, and is now the basis for the associated ratings / limit
+          snapshot.  
+        * *superseded* indicates that the proposal was superseded to a more conservative rating.  This rating may have 
```

Comment:
I think, if I understand what you're saying, the proposal could be viewed as only having two statuses- basically, has gone through the clearing process, or not.  What I'm overloading here is the effective result of the clearing process in the enum.  

I guess what I was thinking here was that it would be good from a REST point of view for one to think about a typical happy path- most of the time, you send in a proposal, and you simply expect it to be accepted.  So, one could see using the API semantically in this way- I was thinking about the RESTBucks example in _REST in Practice_ where the proposal is kind of like a coffee order.  You make a proposal, and with the GET, you check to see what happened to it; this is, as I've read, a good REST design best practice for things where you have an HTTP 202 status.  

You're trying to know if what you sent got accepted.  If for some reason it is NOT accepted, then that is interesting.  You can infer it from looking at the snapshot, but it occurred to me that is, at least potentially, opaque.  Or maybe indirect.  You see a value in the snapshot that is different than what you submitted- so why?  Of course, I've yet to add things like the override reason and limiting branch to the operating snapshot, so yeah you should be able to figure it out.  I was kind of thinking there is value in having this connection be really direct and explicit in the status of the proposal, so you can structurally tie a proposal directly to what happened with it.  What do you think?